proxy/handlers.go

```diff
@@ -351,38 +320,29 @@ func (p *Proxy) UserDashboard(w http.ResponseWriter, r *http.Request) {
 func (p *Proxy) Refresh(w http.ResponseWriter, r *http.Request) {
 	session, err := p.sessionStore.LoadSession(r)
 	if err != nil {
-		httpErr := &httputil.Error{Message: err.Error(), Code: http.StatusBadRequest}
-		httputil.ErrorResponse(w, r, httpErr)
+		httputil.ErrorResponse(w, r, err)
 		return
 	}
 	iss, err := session.IssuedAt()
 	if err != nil {
-		log.FromRequest(r).Error().Err(err).Msg("proxy: couldn't get token's create time")
-		httpErr := &httputil.Error{Code: http.StatusInternalServerError}
-		httputil.ErrorResponse(w, r, httpErr)
+		httputil.ErrorResponse(w, r, err)
 		return
 	}
 
 	// reject a refresh if it's been less than the refresh cooldown to prevent abuse
 	if time.Since(iss) < p.refreshCooldown {
-		log.FromRequest(r).Error().Dur("cooldown", p.refreshCooldown).Err(err).Msg("proxy: refresh cooldown")
-		httpErr := &httputil.Error{
-			Message: fmt.Sprintf("Session must be %v old before refresh", p.refreshCooldown),
-			Code:    http.StatusBadRequest}
-		httputil.ErrorResponse(w, r, httpErr)
+		httputil.ErrorResponse(w, r,
+			httputil.Error(fmt.Sprintf("Session must be %v old before refreshing", p.refreshCooldown), http.StatusBadRequest, nil))
```

Comment:
I feel like we should avoid %v here and cast the interval into a string or integer.